playground/vue-legacy/__tests__/vue-legacy.spec.ts

```diff
@@ -0,0 +1,10 @@
+import { test } from 'vitest'
+import { getBg, untilUpdated } from '~utils'
+
+test('vue legacy assets', async () => {
+  await untilUpdated(() => getBg('.main'), 'assets/asset', true)
+})
+
+test('async vue legacy assets', async () => {
+  await untilUpdated(() => getBg('.module'), 'assets/asset', true)
```

Comment:
About https://github.com/vitejs/vite/pull/9507/commits/9c1a6258afc7976406f7327eec544b82902e414c, `playground/vue-legacy/assets/asset.png` and `playground/vue-legacy/assets/asset2.png` has the same content and we are hitting #8632.